noir-projects/aztec-nr/aztec/src/macros/dispatch.nr

```diff
@@ -9,12 +10,24 @@ pub comptime fn generate_public_dispatch(m: Module) -> Quoted {
 
     let unit = get_type::<()>();
 
+    let seen_selectors =
+        &mut UHashMap::<Field, bool, BuildHasherDefault<Poseidon2Hasher>>::default();
+
     let ifs = functions.map(|function: FunctionDefinition| {
         let parameters = function.parameters();
         let return_type = function.return_type();
 
         let selector: Field = compute_fn_selector(function);
 
+        // Since function selectors are computed as the first 4 bytes of the hash of the function signature,
+        // it's possible to have collisions. With the following assert, we ensure it doesn't happen within
+        // the same contract.
+        assert(
+            !seen_selectors.contains_key(selector),
+            "Public function selector collision detected",
```

Comment:
Can we include the name of the fn that triggered a collision in this msg?

Reply:
Addressed in 

Now the error looks as follows:
<img width="949" alt="image" src="https://github.com/user-attachments/assets/455378fc-3634-4faa-be32-f6fcfccc1e33" />